docs/DifferentiableProgramming.md

```diff
@@ -1539,6 +1645,70 @@ func _<T: FloatingPoint & Differentiable>(x: Tensor<T>, v: Tensor<T>) -> Tensor<
 }
 ```
 
+###### Instance methods
+
+A transpose of a static method is exactly like top-level functions except:
+- When linearity parameters does not include `self`, it must be defined as an
+  instance method in the same type.
```

Comment:
Oh this seems different from the one in [TF-1060](https://bugs.swift.org/browse/TF-1060). Specifically:

2. Instance not wrt self (T, U)
Original: (Self) -> (T, U) -> R
Transpose: (Self.Type) -> (Self, R) -> (T, U)

I thought all methods, no matter if they are being differentiated WRT self, are static. I do like these set of rules better though.

Reply:
Ah, interesting. I don’t know what’s better. Let’s chat in person.

Reply:
There are two options for transpose type calculation not wrt `self`.

```swift
struct S {
  func instanceMethod(_ x: Float) -> Float { x }

  // Question: typing rules for transpose of instance methods not wrt `self`.

  // Option 1: instance method.
  // Pro: more idiomatic.
  // Pro: not consistent with typing rules for instance methods wrt `self`.
  @transpose(of: instanceMethod, wrt: 0)
  func transpose(_ v: Float) -> Float { v }

  // Option 2: static method.
  // Pro: consistent with typing rules for instance methods wrt `self`.
  // Con: `self` parameter for a static method is super weird.
  @transpose(of: instanceMethod, wrt: 0)
  static func transpose(_ `self`: S, _ v: Float) -> Float { v }
}
```

I'd vote for option 1. Option 2 (static method with `self` parameter) is too weird.

Reply:
Right. Option 1 would be my preference too!

Reply:
Agreed I like option 1 as well

Reply:
Should we also add a comment about the param ordering? Specifically how the parameter index must be in increasing order and the `self` needs to be the first in the tuple of `wrt:` params.

Reply:
Done.